pkg/syncer/statussyncer.go

```diff
@@ -66,26 +70,43 @@ func NewStatusSyncer(from, to *rest.Config, syncedResourceTypes []string, kcpClu
 }
 
 func (c *Controller) updateStatusInUpstream(ctx context.Context, gvr schema.GroupVersionResource, upstreamNamespace string, downstreamObj *unstructured.Unstructured) error {
-	upstreamObj := downstreamObj.DeepCopy()
-	upstreamObj.SetUID("")
-	upstreamObj.SetResourceVersion("")
-	upstreamObj.SetNamespace(upstreamNamespace)
+	upstreamStatus := unstructured.Unstructured{}
+	upstreamStatus.SetAPIVersion(downstreamObj.GetAPIVersion())
+	upstreamStatus.SetKind(downstreamObj.GetKind())
+	upstreamStatus.Object["status"] = downstreamObj.Object["status"]
 
-	existing, err := c.toClient.Resource(gvr).Namespace(upstreamNamespace).Get(ctx, upstreamObj.GetName(), metav1.GetOptions{})
+	// Marshalling the unstructured object is good enough as SSA patch
+	data, err := json.Marshal(upstreamStatus.Object)
 	if err != nil {
-		klog.Errorf("Getting resource %s/%s: %v", upstreamNamespace, upstreamObj.GetName(), err)
 		return err
 	}
 
-	// TODO: verify that we really only update status, and not some non-status fields in ObjectMeta.
-	//       I believe to remember that we had resources where that happened.
+	if _, err := c.toClient.Resource(gvr).Namespace(upstreamNamespace).
+		Patch(ctx, downstreamObj.GetName(), types.ApplyPatchType, data, metav1.PatchOptions{
+			FieldManager: syncerApplyManager,
+			Force:        pointer.Bool(true),
+		}, "status"); err != nil {
```

Comment:
nit: Maybe add a blank line to make up for the lack of visual indication between the conditional and its block? Either that or maybe this form of error handling (`if err...; err != nil`) should only apply to single-line conditionals...?

Reply:
I agree, I'd be inclined to go with the "apply to single-line conditionals only". I'll do it if we decide to pursue the work on that PR.